### DISCUSSION TITLE
Bug report - RuntimeError: Unknown qengine

### DISCUSSION THREAD

**Discussion by artem-webdev** [snakers4/silero-models#104] [Q&A] [answered]
## 🐛 Bug

pickle not read model 

## To Reproduce

1. run script for example  - https://github.com/snakers4/silero-models#standalone-use
2.
3.

Traceback (most recent call last):
  File "/Users/ar/PycharmProjects/tts_test-1/main.py", line 13, in <module>
    model = torch.package.PackageImporter(local_file).load_pickle("tts_models", "model")
  File "/Users/ar/PycharmProjects/tts_test-1/venv/lib/python3.9/site-packages/torch/package/package_importer.py", line 249, in load_pickle
    result = unpickler.load()
  File "/usr/local/Cellar/python@3.9/3.9.2_2/Frameworks/Python.framework/Versions/3.9/lib/python3.9/pickle.py", line 1228, in load
    dispatch[key[0]](self)
  File "/usr/local/Cellar/python@3.9/3.9.2_2/Frameworks/Python.framework/Versions/3.9/lib/python3.9/pickle.py", line 1272, in load_binpersid
    self.append(self.persistent_load(pid))
  File "/Users/ar/PycharmProjects/tts_test-1/venv/lib/python3.9/site-packages/torch/package/package_importer.py", line 227, in persistent_load
    loaded_reduces[reduce_id] = func(self, *args)
  File "/Users/ar/PycharmProjects/tts_test-1/venv/lib/python3.9/site-packages/torch/jit/_script.py", line 344, in unpackage_script_module
    cpp_module = torch._C._import_ir_module_from_package(
RuntimeError: Unknown qengine

## Expected behavior

generate voice
## Environment

PyTorch version: 1.10.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 10.13.6 (x86_64)
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.2 (default, Mar 26 2021, 01:41:56)  [Clang 10.0.0 (clang-1000.11.45.5)] (64-bit runtime)
Python platform: macOS-10.13.6-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.3
[pip3] torch==1.10.0
[pip3] torchaudio==0.10.0
[pip3] torchvision==0.11.1
[conda] Could not collect

 - PyTorch Version (e.g., 1.0): 1.10.0
 - OS (e.g., Linux):macos high sierra
 - How you installed PyTorch (`conda`, `pip`, source): pip3 install -q torch torchvision torchaudio omegaconf
 - Build command you used (if compiling from source):
 - Python version: 3.9.2 
 - CUDA/cuDNN version:
 - GPU models and configuration: 2,2 GHz Intel Core i7 4 cores
 - Any other relevant information:

**Comment by snakers4** [comment, marked as answer]
Hi,

> macOS-10.13.6-x86_64-i386-64bit
> OS (e.g., Linux):macos high sierra
> 2,2 GHz Intel Core i7 4 cores

Looks like you are launching the network on a MacBook?

This error typically occurs on older CPUs on old Windows machines. Typically people reported using old Windows versions with CPUs not supporting AVX2 instructions (this is a requirement for  some quantization backends). Some parts of the model are quantized (it was deliberate choice we came to regret, but we wanted to save some space in additional modules).

People reported that reading these links and trying this setting helped:

- https://pytorch.org/docs/stable/quantization.html#introduction-to-quantization
- https://discuss.pytorch.org/t/getting-unknown-qengine-error-when-using-quantized-model-in-a-vm/111283
- `torch.backends.quantized.engine = 'qnnpack'`

Please report if any of this is helpful.

**Reply by lev007-ops** [reply]
> `torch.backends.quantized.engine = 'qnnpack'`

Error
`RuntimeError: quantized engine QNNPACK is not supported`

**Comment by snakers4** [comment]
Also, we have not tested with 1.10 (since it just came out), but my assumption is that it should be compatible.
